Jenkinsfile

```diff
@@ -0,0 +1,43 @@
+pipeline {
+  agent {
+    node {
+      label 'mac-os'
+    }
+    
+  }
+  stages {
+    stage('Build and test') {
+      when {
+        expression {
+          result = sh (script: "git log -1 | grep '.*\\[ci skip\\].*'", returnStatus: true) // Check if commit message contains skip ci label
+          result != 0 // Evaluate the result
+        }
+        
+      }
+      environment {
+        GIT = credentials('65325e52-5ec0-46a7-a937-f81f545f3c1b')
+      }
+      steps {
+        deleteDir()
+        checkout scm
+        dir(path: 'Yona') {
+          sh '/usr/local/bin/pod install'
+          sh 'xcrun agvtool new-marketing-version 1.1'
+          sh 'xcrun agvtool next-version -all'
+          sh 'xcodebuild -allowProvisioningUpdates -workspace Yona.xcworkspace -configuration Debug -scheme Yona archive -archivePath ./BuildOutput/Yona-Debug-$BUILD_NUMBER.xcarchive'
+          sh 'xcodebuild -exportArchive -archivePath ./BuildOutput/Yona-Debug-$BUILD_NUMBER.xcarchive -exportPath ./BuildOutput/Yona-Debug-$BUILD_NUMBER.ipa -exportOptionsPlist ./BuildOutput/ExportOptionsDebug.plist'
+          sh 'xcodebuild -allowProvisioningUpdates -workspace Yona.xcworkspace -configuration Release -scheme Yona archive -archivePath ./BuildOutput/Yona-Release-$BUILD_NUMBER.xcarchive'
+          sh 'xcodebuild -exportArchive -archivePath ./BuildOutput/Yona-Release-$BUILD_NUMBER.xcarchive -exportPath ./BuildOutput/Yona-Release-$BUILD_NUMBER.ipa -exportOptionsPlist ./BuildOutput/ExportOptionsRelease.plist'
+          sh 'git checkout $BRANCH_NAME'
```

Comment:
Why is this necessary? The `checkout scm` command above already checks out the repository

Reply:
Without this check-out, you can't check-in. I faced that for the Android build.

Reply:
configuring the git plugin with to "checkout to local matching branch" seems to work without extra check out step. Other wise the checkout scm is resulting in a detached head.
If we want to continue with detached head branch, we may need to pull the latest changes to it after checkout.

Reply:
Good action. Please update the Android build accordingly.

Reply:
Done. Updated Android pipeline for develop branch. Pipeline executed without any issues after the commit.
I have done that directly without PR, as it is only one line change in jenkins file. Other change is in plugin configuration.

Reply:
Thanks!